### ISSUE TITLE
Kucoin watch_orders does not seem to track order fills correctly

### ISSUE THREAD

**Issue by leastchaos** [ccxt/ccxt#12603]
Hi, I noticed watch_orders does not track order fill properly?
Its a bit difficult to replicate the partial fill.
I think it sometimes misses an order filled update

![image](https://user-images.githubusercontent.com/64965782/161365885-7e9c7238-5692-4364-b4a8-4183ba48fb63.png)
for order id 6247cb3632db5600015b42e2.
There were 2 fills but only the last were shown.
while for order id 6247caaa6ab12d0001e6f3ca. It does show the partial fill.

- OS: linux
- Programming Language version: py
- CCXT version: 1.0.49

```
#pylint: disable=all
import asyncio
import ccxtpro
from pprint import pprint
import sys
# sys.stdout = open('error', 'w')

print(ccxtpro.__version__)
symbol = 'STG/USDT'

exchange: ccxtpro.Exchange = getattr(ccxtpro, 'kucoin')({
    'enableRateLimit': True,
    'apiKey': '',
    'secret': '',
    'password:': '',
    'uid': '',
    'options': {
        'defaultType': 'trade',
    }
})
# exchange.set_sandbox_mode(True)
exchange.verbose=True
print(exchange)
async def main():
    while True:  
        response = await exchange.watch_orders(symbol)
        pprint(response)
    await exchange.close()

asyncio.run(main())
```

[kucoin_watch_orders.txt](https://github.com/ccxt/ccxt/files/8401800/kucoin_watch_orders.txt)

**Comment by leastchaos** [comment]
sorry i just noticed there was a pull request fix i think in 1.0.51. I will update and check

**Comment by leastchaos** [comment]
it seems to be ok now.
